website_blog_group_old_archives/controllers/website_blog.py

```diff
@@ -0,0 +1,75 @@
+from openerp import http, tools
+from openerp.http import request
+from openerp.addons.website_blog.controllers.main import WebsiteBlog
+
+class website_blog(WebsiteBlog):
+
+
+    def nav_list_grouped(self):
+        blog_post_obj = request.registry['blog.post']
+        beginning_of_year = datetime.date.strftime(
+            datetime.date(datetime.date.today().year, 1, 1),
+            tools.DEFAULT_SERVER_DATE_FORMAT
+        )
+        old_groups = blog_post_obj.read_group(
+            request.cr, request.uid, 
+            [['create_date', '<', beginning_of_year]],
+            ['name', 'create_date'],
+            groupby="create_date:year", 
+            orderby="create_date:year desc", 
+            context=request.context
+        )       
+        # the standard orderby is just a comma separated list of fields
+        # optionally followed by desc or asc.
+        # UNDOCUMENTED FEATURE, the time :year, :month, :day work also 
+        # for orderby.
+        new_groups = blog_post_obj.read_group(
+            request.cr, request.uid,
+            [['create_date', '>=', beginning_of_year]],
+            ['name', 'create_date'],
+            groupby="create_date", 
+            orderby="create_date desc", 
+            context=request.context
+        )
+        for group in new_groups:
+            begin_date = datetime.datetime.strptime(
+                group['__domain'][0][2], tools.DEFAULT_SERVER_DATETIME_FORMAT
+            ).date()
+            end_date = datetime.datetime.strptime(
+                group['__domain'][1][2], tools.DEFAULT_SERVER_DATETIME_FORMAT
+            ).date()
+            group['date_begin'] = '%s' % datetime.date.strftime(
+                begin_date, tools.DEFAULT_SERVER_DATE_FORMAT
+            )
+            group['date_end'] = '%s' % datetime.date.strftime(
+                end_date, tools.DEFAULT_SERVER_DATE_FORMAT
+            )
+        for group in old_groups:
+            begin_date = datetime.datetime.strptime(
+                group['__domain'][0][2], tools.DEFAULT_SERVER_DATETIME_FORMAT
+            ).date()
+            end_date = datetime.datetime.strptime(
+                group['__domain'][1][2], tools.DEFAULT_SERVER_DATETIME_FORMAT
+            ).date()
+            group['date_begin'] = '%s' % datetime.date.strftime(
+                begin_date, tools.DEFAULT_SERVER_DATE_FORMAT
+            )
+            group['date_end'] = '%s' % datetime.date.strftime(
+                end_date, tools.DEFAULT_SERVER_DATE_FORMAT
+            )
+        return {'new_groups': new_groups, 'old_groups': old_groups}
+
+    @http.route()
+    def blog(self, blog=None, tag=None, page=1, **opt):
+        result = super(website_blog, self).blog(
+                blog=blog, tag=tag, page=page, **opt
+            )
+        result.qcontext['nav_list_old_grouped'] = self.nav_list_grouped()['old_groups']
+        result.qcontext['nav_list_new_grouped'] = self.nav_list_grouped()['new_groups']
+	return result
```

Comment:
aha! this line uses a tab instead of spaces, that's why it looks weird here. Why this works eludes me to be honest, but I also can't find a definite source on how the python interpreter handles mixed spaces and tabs. Then it's only a violation of the rule to use spaces instead of tabs